src/ralph/backends/database/clickhouse.py

```diff
@@ -269,48 +269,80 @@ def chunk_id_list(chunk_size=10000):
             logger.error("%s. %s", msg, error)
             raise BackendException(msg, *error.args) from error
 
-    def query_statements(self, params: StatementParameters) -> StatementQueryResult:
-        """Returns the results of a statements query using xAPI parameters."""
-        params = asdict(params)
-        where_clauses = []
-
-        if params["statementId"]:
-            where_clauses.append("event_id = {statementId:UUID}")
+    @staticmethod
+    def _add_agent_filters(clickhouse_params, where_clauses, agent_params, qp, tf):
```

Comment:
You can explicit the qp and tf parameters, it would be more convenient for code readability 
```suggestion
    def _add_agent_filters(clickhouse_params, where_clauses, agent_params, query_parameter, target_field):
```

Reply:
I put these as putting the full `query_param` and `target_field` forces additional line breaks which I find ultimately less readable. What do you think ?

```
        if agent_params.mbox:
            clickhouse_params[f"{query_param}__mbox"] = agent_params.mbox
            where_clauses.append(
                f"event.{target_field}.mbox = {{{query_param}__mbox:String}}"
            )

        if agent_params.mbox_sha1sum:
            clickhouse_params[
                f"{query_param}__mbox_sha1sum"
            ] = agent_params.mbox_sha1sum
            where_clauses.append(
                f"event.{target_field}.mbox_sha1sum = {{{query_param}__mbox_sha1sum:String}}"
            )

        if agent_params.openid:
            clickhouse_params[f"{query_param}__openid"] = agent_params.openid
            where_clauses.append(
                f"event.{target_field}.openid = {{{query_param}__openid:String}}"
            )
```

vs

```
        if agent_params.mbox:
            clickhouse_params[f"{qp}__mbox"] = agent_params.mbox
            where_clauses.append(f"event.{tf}.mbox = {{{qp}__mbox:String}}")

        if agent_params.mbox_sha1sum:
            clickhouse_params[f"{qp}__mbox_sha1sum"] = agent_params.mbox_sha1sum
            where_clauses.append(
                f"event.{tf}.mbox_sha1sum = {{{qp}__mbox_sha1sum:String}}"
            )

        if agent_params.openid:
            clickhouse_params[f"{qp}__openid"] = agent_params.openid
            where_clauses.append(f"event.{tf}.openid = {{{qp}__openid:String}}")
```

Reply:
On second thought, your suggestion makes more sense, as mine would add a new pattern in the code. Made the changes :)